src/schema/v2/me/recentlyViewedArtworks.ts

```diff
@@ -17,27 +17,35 @@ export const RecentlyViewedArtworks: GraphQLFieldConfig<
   resolve: async (
     { recently_viewed_artwork_ids },
     args,
-    { artworksLoader, recentlyViewedArtworkIdsLoader, userID }
+    {
+      artworksLoader,
+      recentlyViewedArtworkIdsLoader,
+      userID,
+      xImpersonateUserID,
+    }
   ) => {
-    if (!userID) return null
+    if (!userID && !xImpersonateUserID) return null
+
     const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
 
     try {
-      let recentlyViewedIds = recently_viewed_artwork_ids || []
-      if (!recentlyViewedIds.length) {
-        const recentlyViewedArtworksBody = (
-          await recentlyViewedArtworkIdsLoader(userID)
-        )?.body
-        recentlyViewedIds = recentlyViewedArtworksBody || []
-      }
+      let artworkIDs
 
-      const pageArtworkIDs = recentlyViewedIds.slice(offset, offset + size)
+      // If `recently_viewed_artwork_ids` exists, use those
```

Comment:
The approach here is:

  - if `recently_viewed_artwork_ids` exists (even if it's empty), then we already fetched it via `meLoader` and don't need to fetch anything further
  - otherwise, if impersonation is happening, only then can we fetch the id's specifically

The same approach is taken in the `similarToRecentlyViewed` resolver below